src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -134,7 +134,7 @@ static bool IsRecurringRTE(RangeTblEntry *rangeTableEntry,
 static bool IsRecurringRangeTable(List *rangeTable, RecurringTuplesType *recurType);
 static bool HasRecurringTuples(Node *node, RecurringTuplesType *recurType);
 static bool IsReadIntermediateResultFunction(Node *node);
-static void ValidateClauseList(List *clauseList);
+static DeferredErrorMessage * ValidateClauseList(List *clauseList);
```

Comment:
Minor: We have an unwritten rule of prefixing with `DeferError..`. So maybe `DeferErrorIfNotValidClauseList`?